### PR TITLE
Check Octopoes and Katalogus health before creating orgs, "rollback" on exceptions

### PR DESCRIPTION
**Pull request by Donnype**
## Issue ticket number and link
Closes https://github.com/minvws/nl-kat-coordination/issues/428.

This makes adding organizations through Rocky a more atomic operation to reduce inconsistencies.

### CODE REVIEW

**Comment by Donnype** [comment]
I also added a fix for the bug in `rocky/middleware/onboarding.py`: redteamers might not have an organization member after an organization has been deleted.

**Review by dekkers — CHANGES_REQUESTED** [review]
This looks good to me, but we should also catch RockyError in  OnboardingOrganizationSetupView and return a proper error message to the user. We currently only do this in the admin (see rocky/rocky/admin.py). The saving is done in form_valid, so we should catch the exception when calling super and also add the success notification after saving.

**Comment by Donnype** [comment]
@dekkers See update:

image

File: rocky/tools/models.py [hunk 1]
```diff
@@ -73,42 +75,76 @@ def get_absolute_url(self):
...
+            except Exception as e:
+                raise RockyError(
+                    f"Could not delete organization in the Katalogus after failing to create the "
+                    f"organization in the Katalogus: {e}"
+                ) from e
+
             raise RockyError(f"Octopoes returned error creating organization: {e}") from e
 
+    @staticmethod
+    def _get_healthy_katalogus_and_octopoes(organization_code) -> Tuple[KATalogusClientV1, OctopoesAPIConnector]:
```
Comment (ammar92) [inline rocky/tools/models.py:124]: It makes more sense to split this into two functions

File: rocky/rocky/middleware/onboarding.py [hunk 2]
```diff
@@ -24,12 +24,15 @@ def middleware(request):
                 or request.path.startswith("/api/")
             ):
                 if not member_onboarded:
-                    if is_red_team(request.user):
+                    member = OrganizationMember.objects.filter(user=request.user)
+
+                    # There might be redteamers without an organization after an organization is deleted.
+                    if member.exists() and is_red_team(request.user):  #
```
Comment (ammar92) [inline rocky/rocky/middleware/onboarding.py:30]:
```suggestion
                    if member.exists() and is_red_team(request.user):
```

**Review by ammar92 — COMMENTED** [review]
Looks good, some remarks/ questions

File: rocky/rocky/middleware/onboarding.py (hunk 2 above)
Comment (underdarknl) [reply rocky/rocky/middleware/onboarding.py:30]: This is not just limited to red-teamers. Admins and clients are mostly the same.

Comment (Donnype) [reply rocky/rocky/middleware/onboarding.py:30]: @Rieven Was there a specific reason to filter on redteamers here?

Comment (Rieven) [reply rocky/rocky/middleware/onboarding.py]: > @Rieven Was there a specific reason to filter on redteamers here?

@Donnype Yes, when member is not onboarded at startup it is redirected to the redteamers onboarding flow/views.

Comment (Donnype) [reply rocky/rocky/middleware/onboarding.py]: Answer: yes clients do not necessarily need to be onboarded(?)

**Comment by Darwinkel** [comment]
## Checklist for functional reviewer(s):
- [x] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [x] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [x] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* No functional regressions under normal circumstances
* Proper error message when Octopoes is down
* Proper error message when Katalogus is down

Nice work! I think this may actually be extremely helpful in debugging what exactly goes wrong in the CI's Robot tests!

**Review by Darwinkel — APPROVED** [review]

**Pull request merged**